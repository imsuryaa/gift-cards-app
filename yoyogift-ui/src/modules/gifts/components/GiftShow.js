import React from "react";
import Styles from "../../../assets/css/GiftShow.module.css";
import Grid from "@material-ui/core/Grid";
import { DateFormatter } from "./../../common/components/DateFormatter";
import StarRatingComponent from "react-star-rating-component";
import SendGiftCardDialog from "../../common/components/DraggableDialog";

const GiftShow = props => {
  const { data } = props;

  return (
    <div style={{ padding: "4%" }}>
      <div className={Styles.detailsPage}>
        <Grid container spacing={0}>
          <Grid item xs={12} sm={5} className={Styles.cardContainer}>
            <div>
              <Grid container spacing={0}>
                <Grid item xs={6} sm={6}>
                  <img
                    src={data.cardImage}
                    alt="vendorImage"
                    className={Styles.vendorImage}
                  />
                </Grid>
                <Grid item xs={6} sm={6}>
                  <p className={Styles.cardName}>{data.cardName}</p>
                  <span style={{marginLeft:'30%'}}><label className={Styles.label}>Points:</label>{data.cardPoints}</span> 
                </Grid>
              </Grid>
            </div>
            <SendGiftCardDialog getEmail={(email) => {props.getEmail(email)}}/>
            <p><label className={Styles.label}>NUMBER OF CARDS AVAILABLE: </label>{data.cardCount}</p>
            <p><label className={Styles.label}>EXPIRY DATE: </label>{DateFormatter(data.cardExpiryDate)}</p>
            <div>
              <p className={Styles.label}> RATING & REVIEWS:</p>
              {
                data.cardComments ? 
                data.cardComments.map(comment => {
                return (
                  <div className={Styles.commentsSection}>
                    <p className={Styles.commentorName}> {comment.first_name} {comment.last_name}<span className={Styles.commentDate}>{DateFormatter(comment.commented_on)}</span></p>
                    <p className={Styles.comment}>{comment.comment}</p>
                    <StarRatingComponent
                        name="rating"
                        starCount={5}
                        value={comment.rating}
                        editing={false}
                      />
                  </div>
                )
              })
            :  <div>Loading User Reviews..</div>}
            </div>
          </Grid>
          <Grid item xs={12} sm={6} className={Styles.detailsContainer}>
            <p className={Styles.vendorName}>{data.cardVendor}</p>
            <p className={Styles.cardDescription}>{data.cardLongDesc}</p>
            {/* <p className={Styles.cardDescription}>Named “America’s Healthiest Grocery Store” by Health Magazine, Whole Foods Market is the leading organic and natural food retailer, prioritizing animal welfare, seafood sustainability, organic agriculture, and safer personal and home care products. Whole Foods Market strives to provide customers with a superiorcustomer service experience, while also supporting local producers, bettering the communities in which it operates, and improving the environment. Because of its dedication to strict quality standards and excellence, Whole Foods Market is able to offer its corporate and individual customers alike a consistently exceptional shopping experience. Whole Foods Market customers shop for quality and consistency, and are very loyal to the brands they like, patronizing the same stores again and again. Less focused on the bargain, Whole Foods Market customers have expendable income and care about where they spend their money, even going out of their way to shop at preferred retailers. Gift cards never expire and can buy anything at any of our 400 stores in the US and Canada. For more information on where to find the closest store</p> */}
          </Grid>
        </Grid>
      </div>
    </div>
  );
};
export default GiftShow;
