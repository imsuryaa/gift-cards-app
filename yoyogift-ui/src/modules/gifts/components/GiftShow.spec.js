import React from "react";
import { mount } from "enzyme";

import GiftShow from "./GiftShow";

describe("Gift Show Test Suite", () => {
  it('should render the expected HTML', () => {
    const data = {
      id: 1,
      cardName: "Amazon Gift SAS UPDATED",
      cardPoints: "1200",
      cardCategory: "Ecommerce",
      cardRetailer: "Amazon",
      cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
      cardExpiryDate: "2019-06-29",
      cardCount: 86,
      cardImage: "https://pngimg.com/uploads/amazon/amazon_PNG21.png",
      cardVendor: "amazon",
      cardShortDesc: "10% OFF",
      cardLongDesc: "Amazon Gift Cards are the Perfect Gift, Every Time. Use the eBay Gift Card to shop from millions of items in Electronics, Toys, Motors, Fashion, Home & Garden, Art, Collectibles, Sporting Goods and everything in-between. eBay Gift Cards never expire and have no fees. Use it to shop now or wait for the deal of a lifetime.",
      cardComments: [
        {
          id: 1,
          first_name: "Sebastian",
          last_name: "Eschweiler",
          email: "sebastian@mindtree.com",
          rating: 4,
          comment: "Great gift card. Happy to gift!",
          commented_on: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        },
        {
          id: 2,
          first_name: "Bill",
          last_name: "Joe",
          email: "billjoe@mindtree.com",
          rating: 3,
          comment: "10% discount on Amazon. Yay",
          commented_on: "Sun Jun 12 2019 15:43:25 GMT+0530 (India Standard Time)"
        }
      ]
    };
    expect(mount(<GiftShow getEmail={jest.fn()} data={data}/>
    ).html()).toMatchSnapshot();
  });
});