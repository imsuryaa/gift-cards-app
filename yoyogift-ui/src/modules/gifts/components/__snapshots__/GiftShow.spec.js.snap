// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Show Test Suite should render the expected HTML 1`] = `"<div style=\\"padding: 4%;\\"><div class=\\"detailsPage\\"><div class=\\"MuiGrid-container-1\\"><div class=\\"MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-5-48 cardContainer\\"><div><div class=\\"MuiGrid-container-1\\"><div class=\\"MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-6-49\\"><img src=\\"https://pngimg.com/uploads/amazon/amazon_PNG21.png\\" alt=\\"vendorImage\\" class=\\"vendorImage\\"></div><div class=\\"MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-6-49\\"><p class=\\"cardName\\">Amazon Gift SAS UPDATED</p><span style=\\"margin-left: 30%;\\"><label class=\\"label\\">Points:</label>1200</span></div></div></div><div><button class=\\"MuiButtonBase-root-124 MuiButton-root-98 MuiButton-contained-109 MuiButton-containedPrimary-110 MuiButton-raised-112 MuiButton-raisedPrimary-113\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-99\\">send this gift card</span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div><p><label class=\\"label\\">NUMBER OF CARDS AVAILABLE: </label>86</p><p><label class=\\"label\\">EXPIRY DATE: </label>2019-06-29</p><div><p class=\\"label\\"> RATING &amp; REVIEWS:</p><div class=\\"commentsSection\\"><p class=\\"commentorName\\"> Sebastian Eschweiler<span class=\\"commentDate\\">2019-05-19</span></p><p class=\\"comment\\">Great gift card. Happy to gift!</p><div style=\\"display: inline-block; position: relative;\\" class=\\"dv-star-rating dv-star-rating-non-editable\\"><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_5\\" value=\\"5\\"><label style=\\"float: right; cursor: default; color: rgb(51, 51, 51);\\" class=\\"dv-star-rating-star dv-star-rating-empty-star\\" for=\\"rating_5\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_4\\" value=\\"4\\" checked=\\"\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_4\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_3\\" value=\\"3\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_3\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_2\\" value=\\"2\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_2\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_1\\" value=\\"1\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_1\\"><i style=\\"font-style: normal;\\">★</i></label></div></div><div class=\\"commentsSection\\"><p class=\\"commentorName\\"> Bill Joe<span class=\\"commentDate\\">2019-06-12</span></p><p class=\\"comment\\">10% discount on Amazon. Yay</p><div style=\\"display: inline-block; position: relative;\\" class=\\"dv-star-rating dv-star-rating-non-editable\\"><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_5\\" value=\\"5\\"><label style=\\"float: right; cursor: default; color: rgb(51, 51, 51);\\" class=\\"dv-star-rating-star dv-star-rating-empty-star\\" for=\\"rating_5\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_4\\" value=\\"4\\"><label style=\\"float: right; cursor: default; color: rgb(51, 51, 51);\\" class=\\"dv-star-rating-star dv-star-rating-empty-star\\" for=\\"rating_4\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_3\\" value=\\"3\\" checked=\\"\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_3\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_2\\" value=\\"2\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_2\\"><i style=\\"font-style: normal;\\">★</i></label><input style=\\"display: none; position: absolute; margin-left: -9999px;\\" class=\\"dv-star-rating-input\\" type=\\"radio\\" name=\\"rating\\" id=\\"rating_1\\" value=\\"1\\"><label style=\\"float: right; cursor: default; color: rgb(255, 180, 0);\\" class=\\"dv-star-rating-star dv-star-rating-full-star\\" for=\\"rating_1\\"><i style=\\"font-style: normal;\\">★</i></label></div></div></div></div><div class=\\"MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 detailsContainer\\"><p class=\\"vendorName\\">amazon</p><p class=\\"cardDescription\\">Amazon Gift Cards are the Perfect Gift, Every Time. Use the eBay Gift Card to shop from millions of items in Electronics, Toys, Motors, Fashion, Home &amp; Garden, Art, Collectibles, Sporting Goods and everything in-between. eBay Gift Cards never expire and have no fees. Use it to shop now or wait for the deal of a lifetime.</p></div></div></div></div>"`;
